app/controllers/admin/appointments_controller.rb

```diff
@@ -37,8 +37,15 @@ def create
         @appointment = Appointment.new(create_params)
         @appointment.end = @appointment.start + @appointment.ubs.slot_interval_minutes.minutes
         vaccine = Vaccine.find_by id: params[:vaccine_id]
+        last_dose = @appointment.patient.doses.order(:sequence_number).last
+        patient = @appointment.patient
 
-        if vaccine && @appointment.save
+        if @appointment.patient.doses.exists? && @appointment.save
```

Comment:
I think this check should come before line #40, because if `doses` is empty, line #40 is gonna break

Reply:
Of course, this change had been made in past commit and I ended up mistakenly changing.